packages/ecs/src/schemas/JSONSchemas.ts

```diff
@@ -262,74 +223,21 @@ export const S = {
   Func: <Params extends Schema[], Return extends Schema, Initial extends (...params: any[]) => any>(
     parameters: [...Params],
     returns: Return,
-    init?: Initial,
     options?: TFuncSchema<Params, Return>['options']
   ) =>
     ({
-      [Kind]: 'Func',
-      options: buildOptions(() => init, options),
+      ...buildSchema('Func', options),
       properties: { params: parameters, return: returns }
     }) as TFuncSchema<Params, Return>,
 
-  Call: <Initial extends (...params: any[]) => any>(
-    init?: Initial,
-    options?: TFuncSchema<Schema[], TVoidSchema>['options']
-  ) => S.Func([], S.Void(), init, options),
-
-  /**
-   * Schemas wrapped in this schema may be null, will default to null if not default value is provided
-   * This should only be needed for properties of a component that are fetched asynchronously or provided by a third party api
-   */
-  Nullable: <T extends Schema, Initial>(
-    schema: T,
-    init?: Initial,
-    options?: TUnionSchema<[T, TNullSchema]>['options']
-  ) => S.Union([S.Null(), schema], init ?? null, options),
+  Call: <Initial extends (...params: any[]) => any>(options?: TFuncSchema<Schema[], TVoidSchema>['options']) =>
+    S.Func([], S.Void(), options),
 
   /**
    * Schemas wrapped in this schema are optional values that can be undefined, will default to undefined if not default value is provided
    */
-  Optional: <T extends Schema, Initial>(
-    schema: T,
-    init?: Initial,
-    options?: TUnionSchema<[T, TUndefinedSchema]>['options']
-  ) => S.Union([S.Undefined(), schema], init ?? undefined, options),
-
-  /**
-   *
-   * Schemas wrapped in this schema are required to be provided when a component is set or will throw an error
-   *
-   * Throws error
-   * S.Object({ test: S.Required(S.Number(0)) })
-   * setComponent(entity, Component, {})
-   *
-   * No error thrown
-   * S.Object({ test: S.Required(S.Number(0)) })
-   * setComponent(entity, Component, {test: 0})
-   *
-   */
-  Required: <T extends Schema, Initial>(schema: T, init?: Initial, options?: TRequiredSchema<T>['options']) =>
-    ({
-      [Kind]: 'Required',
-      options: buildOptions(init, options),
-      properties: schema
-    }) as TRequiredSchema<T>,
-
-  /**
-   *
-   * Schemas wrapped in this schema will be ignored during serialization
-   *
-   * S.Object({ test: S.Number(0) }) serializes to { test: 0 }
-   * S.Object({ test: S.NonSerialized(S.Number(0)) }) serializes to {}
-   * S.NonSerialized(S.Object({ test: S.Number(0) })) serializes to null
-   *
-   */
-  NonSerialized: <T extends Schema, Initial>(schema: T, init?: Initial, options?: TNonSerializedSchema<T>['options']) =>
-    ({
-      [Kind]: 'NonSerialized',
-      options: buildOptions(init, options),
-      properties: schema
-    }) as TNonSerializedSchema<T>,
+  Optional: <T extends Schema>(schema: T, options?: TUnionSchema<[T, TUndefinedSchema]>['options']) =>
```

Comment:
would we be able to remove the S.Undefined type and use an `{ optional: true }` option field?